notification-config-service-impl/src/main/java/org/hypertrace/notification/config/service/NotificationConfigServiceImpl.java

```diff
@@ -0,0 +1,230 @@
+      StreamObserver<UpdateNotificationRuleResponse> responseObserver) {
+    try {
+      RequestContext requestContext = RequestContext.CURRENT.get();
+      notificationConfigServiceRequestValidator.validateUpdateNotificationRuleRequest(
+          requestContext, request);
+      responseObserver.onNext(
+          UpdateNotificationRuleResponse.newBuilder()
+              .setNotificationRule(
+                  notificationRuleStore.upsertObject(requestContext, request.getNotificationRule()))
+              .build());
+      responseObserver.onCompleted();
+    } catch (Exception e) {
+      log.error("Update Notification Rule RPC failed for request:{}", request, e);
+      responseObserver.onError(e);
+    }
+  }
+
+  @Override
+  public void getAllNotificationRules(
+      GetAllNotificationRulesRequest request,
+      StreamObserver<GetAllNotificationRulesResponse> responseObserver) {
+    try {
+      RequestContext requestContext = RequestContext.CURRENT.get();
+      notificationConfigServiceRequestValidator.validateGetAllNotificationRulesRequest(
+          requestContext, request);
+      responseObserver.onNext(
+          GetAllNotificationRulesResponse.newBuilder()
+              .addAllNotificationRules(notificationRuleStore.getAllObjects(requestContext))
+              .build());
+      responseObserver.onCompleted();
+    } catch (Exception e) {
+      log.error("Get All Notification Rules RPC failed for request:{}", request, e);
+      responseObserver.onError(e);
+    }
+  }
+
+  @Override
+  public void deleteNotificationRule(
+      DeleteNotificationRuleRequest request,
+      StreamObserver<DeleteNotificationRuleResponse> responseObserver) {
+    try {
+      RequestContext requestContext = RequestContext.CURRENT.get();
+      notificationConfigServiceRequestValidator.validateDeleteNotificationRuleRequest(
+          requestContext, request);
+      notificationRuleStore.deleteObject(requestContext, request.getNotificationRuleId());
+      responseObserver.onNext(DeleteNotificationRuleResponse.getDefaultInstance());
+      responseObserver.onCompleted();
+    } catch (Exception e) {
+      log.error("Delete Notification Rule RPC failed for request:{}", request, e);
+      responseObserver.onError(e);
+    }
+  }
+
+  @Override
+  public void createNotificationChannel(
+      CreateNotificationChannelRequest request,
+      StreamObserver<CreateNotificationChannelResponse> responseObserver) {
+    try {
+      RequestContext requestContext = RequestContext.CURRENT.get();
+      notificationConfigServiceRequestValidator.validateCreateNotificationChannelRequest(
+          requestContext, request);
+      NotificationChannelMutableData notificationChannelMutableData =
+          request.getNewNotificationChannel().getNotificationChannelMutableData();
+      NotificationChannel.Builder builder =
+          NotificationChannel.newBuilder()
+              .setId(UUID.randomUUID().toString())
```

Comment:
Nit - we have a uuidgenerator util too (useful for testing to inject rather than use a static)

Reply:
that is not present in ht ig

Reply:
ah so it's not. I'll promote it when I get a chance.